### PR TITLE
Solvers dox

### PR DESCRIPTION
**Pull request by acfisher**
Improve the doxygen for various solvers.

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge|
 | --- | --- | --- | --- | --- | --- | --- |
| [#4053](https://github.com/mfem/mfem/pull/4053) | @acfisher | @tzanio | @vladotomov + @paul-hilscher + @mlstowell | 1/16/24 | 2/3/24 | 2/5/24 | |

<details>
<summary>PR Checklist</summary>
    
- [ ] Code builds.
- [ ] Code passes `make style`.
- [ ] Update `CHANGELOG`:
    - [ ] Is this a new feature users need to be aware of? New or updated example or miniapp?
    - [ ] Does it make sense to create a new section in the `CHANGELOG` to group with other related features?
- [ ] Update `INSTALL`:
    - [ ] Had a new optional library been added? If so, what range of versions of this library are required? (*Make sure the external library is compatible with our BSD license, e.g. it is not licensed under GPL!*)
    - [ ] Have the version ranges for any required or optional libraries changed?
    - [ ] Does `make` or `cmake` have a new target?
    - [ ] Did the requirements or the installation process change? *(rare)*
- [ ] Update continuous integration server configurations if necessary (e.g. with new version requirements for each of MFEM's dependencies)
    - [ ] `.github`
    - [ ] `.appveyor.yml`
- [ ] Update `.gitignore`:
    - [ ] Check if `make distclean; git status` shows any files that were generated from the source by the project (not an IDE) but we don't want to track in the repository.
    - [ ] Add new patterns (just for the new files above) and re-run the above test.
- [ ] New examples:
    - [ ] All sample runs at the top of the example source file work.
    - [ ] Update `examples/makefile`:
      - [ ] Add the example code to the appropriate `SEQ_EXAMPLES` and `PAR_EXAMPLES` variables.
      - [ ] Add any files generated by it to the `clean` target.
      - [ ] Add the example binary and any files generated by it to the top-level `.gitignore` file.
    - [ ] Update `examples/CMakeLists.txt`:
      - [ ] Add the example code to the `ALL_EXE_SRCS` variable.
      - [ ] Make sure `THIS_TEST_OPTIONS` is set correctly for the new example.
   - [ ] List the new example in `doc/CodeDocumentation.dox`.
   - [ ] If new examples directory (e.g.`examples/pumi`), list it in `doc/CodeDocumentation.conf.in`
   - [ ] Companion pull request for documentation in [mfem/web](https://github.com/mfem/web) repo:
      - [ ] Update or add example-specific documentation, see e.g. the `src/examples.md`.
      - [ ] Add the description, labels and screenshots in `src/examples.md` and `src/img`.
      - [ ] In `examples.md`, list the example under the appropriate categories, add new categories if necessary.
      - [ ] Add a short description of the example in the "Extensive Examples" section of `features.md`.
- [ ] New miniapps:
   - [ ] All sample runs at the top of the miniapp source file work.
   - [ ] Update top-level `makefile` and `makefile` in corresponding miniapp directory.
   - [ ] Add the miniapp binary and any files generated by it to the top-level `.gitignore` file.
   - [ ] Update CMake build system:
      - [ ] Update the `CMakeLists.txt` file in the `miniapps` directory, if the new miniapp is in a new directory.
      - [ ] Add/update the `CMakeLists.txt` file in the new miniapp directory.
      - [ ] Consider adding a new test for the new miniapp.
   - [ ] List the new miniapp in `doc/CodeDocumentation.dox`
   - [ ] If new miniapps directory (e.g.`miniapps/nurbs`), add it to `MINIAPP_SUBDIRS` in the `makefile`.
   - [ ] If new miniapps directory (e.g.`miniapps/nurbs`), list it in `doc/CodeDocumentation.conf.in`
   - [ ] Companion pull request for documentation in [mfem/web](https://github.com/mfem/web) repo:
     - [ ] Update or add miniapp-specific documentation, see e.g. the `src/meshing.md` and `src/electromagnetics.md` files.
     - [ ] Add the description, labels and screenshots in `src/examples.md` and `src/img`.
     - [ ] The miniapps go at the end of the page, and are usually listed only under a specific "Application (PDE)" category.
     - [ ] Add a short description of the miniapp in the "Extensive Examples" section of `features.md`.
- [ ] New capability:
   - [ ] All new public, protected, and private classes, methods, data members, and functions have full Doxygen-style documentation in source comments. Documentation should include descriptions of member data, function arguments and return values, template parameters, and prerequisites for calling new functions.
   - [ ] Pointer arguments and return values must specify whether ownership is being transferred or lent with the call.
   - [ ] Any new functions should include descriptions of their intended use e.g. for internal use only, user-facing, etc., along with references to example code whenever possible/appropriate.
   - [ ] Consider adding new sample runs in existing examples to highlight the new capability.
   - [ ] Consider saving cool simulation pictures with the new capability in the Confluence gallery (LLNL only) or submitting them, via pull request, to the gallery section of the `mfem/web` repo.
   - [ ] If this is a major new feature, consider mentioning it in the short summary inside `README` *(rare)*.
   - [ ] List major new classes in `doc/CodeDocumentation.dox` *(rare)*.
- [ ] Update this checklist, if the new pull request affects it.
- [ ] Run `make unittest` to make sure all unit tests pass.
- [ ] Run the tests in `tests/scripts`.
- [ ] (LLNL only) After merging:
   - [ ] Update internal tests to include the new features.
</details>

### CODE REVIEW

File: linalg/amgxsolver.hpp [hunk 1]
```diff
@@ -71,85 +71,120 @@ class AmgXSolver : public Solver
...
+      verbosity. Assumes no MPI parallelism.
    */
    AmgXSolver(const AMGX_MODE amgxMode_, const bool verbose);
 
-   /**
-      Once the solver configuration has been established through either the
-      ReadParameters method or the constructor, InitSerial will initialize the
-      library. If configuring with constructor, the constructor will make this
-      call.
+   /** @brief Initilize the AmgX library for serial execution once
```
Comment (paul-hilscher) [inline linalg/amgxsolver.hpp:110]: minor: spelling. "Initilize" -> "Initialize"

File: linalg/amgxsolver.hpp [hunk 2]
```diff
@@ -71,85 +71,120 @@ class AmgXSolver : public Solver
...
 
-   /**
-      Replaces the matrix coefficients in the AmgX solver.
+   /** @brief Change the input operator that is being solved via AmgX.
+      Supports operators based on either an MFEM SparseMatrix or
+      HypreParMatrix.
    */
    void UpdateOperator(const Operator &op);
 
+   /** @brief Untilize the AmgX library to solve the linear system
```
Comment (paul-hilscher) [inline linalg/amgxsolver.hpp:167]: minor: spelling "Untilize" -> "Utilize" ?

File: linalg/amgxsolver.hpp [hunk 3]
```diff
@@ -71,85 +71,120 @@ class AmgXSolver : public Solver
...
    */
    AmgXSolver(const MPI_Comm &comm, const int nDevs,
               const AMGX_MODE amgx_Mode_, const bool verbose);
 
-   /**
-      Once the solver configuration has been established, either through the
-      constructor or the ReadParameters method, InitSerial will initialize the
-      library. If configuring with constructor, the constructor will make this
-      call.
+   /** @brief Initilize the AmgX library in parallel mode with exactly one
```
Comment (paul-hilscher) [inline linalg/amgxsolver.hpp:137]: minor: spelling. "Initilize" -> "Initialize" ?

File: linalg/amgxsolver.hpp [hunk 4]
```diff
@@ -71,85 +71,120 @@ class AmgXSolver : public Solver
...
+       in will be used.
+   */
    virtual void Mult(const Vector& b, Vector& x) const;
 
+   /// Return the number of iterations that were executed during the last solve phase.
    int GetNumIterations();
 
+   /** @brief Read in the AMGx parameters either through a file or directly through a
+       properly formated string.  If @a source is set to AmgXSolver::EXTERNAL
+       the parameters are loaded from a filename set by @a config.  If If @a source is set
```
Comment (paul-hilscher) [inline linalg/amgxsolver.hpp:181]: minor: "If If" -> "If"

File: linalg/mumps.hpp [hunk 5]
```diff
@@ -34,22 +34,35 @@ namespace mfem
...
 {
 public:
+   /// Specify the type of matrix we are applying the solver to
    enum MatType
    {
+      /// General sparse matrix, no symmetry is assumed
       UNSYMMETRIC = 0,
+      /// A sparse symmetric positive definite matrix
       SYMMETRIC_POSITIVE_DEFINITE = 1,
+      /// A sparse symmetric matrix that is no necissisarilty positive definite
```
Comment (paul-hilscher) [inline linalg/mumps.hpp:44]: minor: spelling "necissisarilty" -> "necessarily" ?

File: linalg/slepc.hpp [hunk 6]
```diff
@@ -51,16 +51,22 @@ class SlepcEigenSolver
...
+      @note Default value is 1e-8
+   */
    void SetTol(double tol);
 
-   /// Set maximum number of iterations
+   /// Set maximum number of iterations allowed in the call to SlepcEigenSolver::Solve
    void SetMaxIter(int max_iter);
-   /// Set the number of required eigenmodes
+
+   /// Set the number of eignemodes to compute
```
Comment (paul-hilscher) [inline linalg/slepc.hpp:63]: minor: spelling "eignemodes" -> "eigenmodes"

File: linalg/slepc.hpp [hunk 7]
```diff
@@ -69,39 +75,92 @@ class SlepcEigenSolver
...
+   */
    enum SpectralTransformation
    {
+      /// Utilize the shift of origin strategy
       SHIFT,
+      /// Utilize the shift and invert strategy
       SHIFT_INVERT
    };
 
+   /** @brief Set the which eignevalues the solver will target and the order they will be indexed in
```
Comment (paul-hilscher) [inline linalg/slepc.hpp:149]: minor:"
- "Set the which" -> "Set which" ?
- spelling: "eignevalues" -> "eigenvalues"

File: linalg/solvers.hpp [hunk 8]
```diff
@@ -338,7 +343,10 @@ class OperatorJacobiSmoother : public Solver
    /// Replace diagonal entries with their absolute values.
    void SetPositiveDiagonal(bool pos_diag = true) { use_abs_diag = pos_diag; }
 
+   /// Approach the solution of the linear system by applying jacobi smoothing
    void Mult(const Vector &x, Vector &y) const;
+
+   /// Approach the solition of the transposed linear system by applying jacobi smoothing
```
Comment (paul-hilscher) [inline linalg/solvers.hpp:349]: minor: spelling: "solution" -> "solution"

File: linalg/solvers.hpp [hunk 9]
```diff
@@ -555,6 +568,7 @@ class FGMRESSolver : public IterativeSolver
 
    void SetKDim(int dim) { m = dim; }
 
+   /// Iterative solution of the linear system using the FGMRESt method
```
Comment (paul-hilscher) [inline linalg/solvers.hpp:571]: minor: "FGMRESt" is the "t" intended  or should this be "FGMRES" ?

File: linalg/solvers.hpp [hunk 10]
```diff
@@ -621,6 +636,7 @@ class MINRESSolver : public IterativeSolver
 
    virtual void SetOperator(const Operator &op);
 
+   /// Iterative solution of the linear system using the Minres method
```
Comment (paul-hilscher) [inline linalg/solvers.hpp:639]: pedantic: but for consistency with comment on l. 617 and "GMRES", "Minres" should be "MINRES"

File: linalg/strumpack.hpp [hunk 11]
```diff
@@ -41,20 +43,26 @@ class STRUMPACKRowLocMatrix : public Operator
...
    void Mult(const Vector &x, Vector &y) const
    {
       MFEM_ABORT("STRUMPACKRowLocMatrix::Mult: Matrix vector products are not "
                  "supported!");
    }
 
+   /// Get the MPI Comm being used by the parallel matrix
    MPI_Comm GetComm() const { return A_->comm(); }
 
+   /// Gain access to the internal CSR matrix
```
Comment (paul-hilscher) [inline linalg/strumpack.hpp:65]: pedantic: Maybe "Gain" -> "Get" access to be consistent with the function name "GetA"

File: linalg/strumpack.hpp [hunk 12]
```diff
@@ -72,138 +80,192 @@ template <typename STRUMPACKSolverType>
...
-   // Set the operator.
+   /** @brief Set the operator/matrix.
+       \note  @a A must be a STRUMPACKRowLocMatrix. */
    void SetOperator(const Operator &op);
 
-   // Set various solver options. Refer to STRUMPACK documentation for
-   // details.
+   /** @brief Set options that were captured from the command line.
+
+      These were captured in the constructer STRUMPACKSolverBase.  Refer
```
Comment (paul-hilscher) [inline linalg/strumpack.hpp:113]: pedantic: Looks like there are two whitespaces before "Refer"

File: linalg/strumpack.hpp [hunk 13]
```diff
@@ -72,138 +80,192 @@ template <typename STRUMPACKSolverType>
...
+    *
     * STRUMPACK is an (approximate) direct solver. It can be used as a direct
     * solver or as a preconditioner. To use STRUMPACK as only a preconditioner,
     * set the Krylov solver to DIRECT. STRUMPACK also provides iterative solvers
     * which can use the preconditioner, and these iterative solvers can also be
     * used without preconditioner.
     *
     * Supported values are:
-    *    AUTO:           Use iterative refinement if no HSS compression is
+    *  - AUTO:           Use iterative refinement if no HSS compression is
```
Comment (paul-hilscher) [inline linalg/strumpack.hpp:159]: minor: "GMRes" -> "GMRES" in this section for consistency

File: linalg/strumpack.hpp [hunk 14]
```diff
@@ -72,138 +80,192 @@ template <typename STRUMPACKSolverType>
...
    void SetCompressionAbsTol(double atol);
+
 #if STRUMPACK_VERSION_MAJOR >= 5
+   /** @brief Set the precision for the lossy compression option
+    *
+    * Use STRUMPACKSolverBase::SetCompression to set the proper compression type.
+    */
    void SetCompressionLossyPrecision(int precision);
+
+   /** @brief Set the number of butterflylevels for the HODLR compression option
```
Comment (paul-hilscher) [inline linalg/strumpack.hpp:260]: minor:  butterflylevels -> "butterfly levels" (e.g. two words in https://petsc.org/release/manualpages/Mat/MatSTRUMPACKSetCompButterflyLevels/)

File: linalg/strumpack.hpp [hunk 15]
```diff
@@ -223,21 +285,29 @@ class STRUMPACKSolver :
...
+
+      SetFromCommandLine must be called for the command line arguments
+      to be used.
+   */
    STRUMPACKSolver(MPI_Comm comm, int argc, char *argv[]);
    MFEM_DEPRECATED STRUMPACKSolver(int argc, char *argv[], MPI_Comm comm)
       : STRUMPACKSolver(comm, argc, argv) {}
 
-   // Constructor with STRUMPACK matrix object and command line arguments.
+   /** @brief Constructor with MSTRUMPACK matrix object and command line arguments.
```
Comment (paul-hilscher) [inline linalg/strumpack.hpp:303]: is the "M" in "MSTRUMPACK" intended ?

File: linalg/strumpack.hpp [hunk 16]
```diff
@@ -247,20 +317,28 @@ class STRUMPACKMixedPrecisionSolver :
...
-   // Constructor with MPI_Comm parameter and command line arguments.
+   /** @brief Constructor with MPI_Comm parameter and command line arguments.
+
+      SetFromCommandLine must be called for the command line arguments
+      to be used.
+   */
    STRUMPACKMixedPrecisionSolver(MPI_Comm comm, int argc, char *argv[]);
 
-   // Constructor with STRUMPACK matrix object and command line arguments.
+   /** @brief Constructor with MSTRUMPACK matrix object and command line arguments.
```
Comment (paul-hilscher) [inline linalg/strumpack.hpp:333]: is the "M" in "MSTRUMPACK" intended ?

**Comment by paul-hilscher** [comment]
Thanks @acfisher for the nice improvements for the doc. I have added a couple of minor comments. Once addressed I will approve.

**Comment by acfisher** [comment]
@paul-hilscher Thank you for the through review and catching my typos.  I'll fix them today and ping you when they are updated.

**Comment by acfisher** [comment]
@paul-hilscher Alright I addressed all of the things you found and did a little cleanup as well.  @mlstowell, can you have a look at this sometime this week?

File: linalg/amgxsolver.hpp [hunk 17]
```diff
@@ -71,85 +71,122 @@ class AmgXSolver : public Solver
...
 
-   /**
+   /** @brief
       Flags to determine whether user solver settings are defined internally in
       the source code or will be read through an external JSON file.
    */
-   enum CONFIG_SRC {INTERNAL, EXTERNAL, UNDEFINED};
+   enum CONFIG_SRC
+   {
+      /// Configuration with be read directly from a string
```
Comment (paul-hilscher) [inline linalg/amgxsolver.hpp:95]: minor: "with be" -> "will be"

File: linalg/amgxsolver.hpp [hunk 18]
```diff
@@ -71,85 +71,122 @@ class AmgXSolver : public Solver
...
    */
    void InitExclusiveGPU(const MPI_Comm &comm);
 
-   /**
-      Once the solver configuration has been established, either through the
-      ReadParameters method, InitMPITeams will initialize the library and create
-      MPI teams based on the number of devices on each node (nDevs). If
-      configuring with constructor, the constructor will make this call.
+   /** @brief Initialize the AmgX library and create MPI teams based on the
+      numberof devices on each node @a nDevs.  If configuring with a
```
Comment (paul-hilscher) [inline linalg/amgxsolver.hpp:147]: minor: "numberof" -> "number of"

File: linalg/amgxsolver.hpp [hunk 19]
```diff
@@ -71,85 +71,122 @@ class AmgXSolver : public Solver
...
+      constructor, theconstructor will make this call, otherwise this will need
```
Comment (paul-hilscher) [inline linalg/amgxsolver.hpp:148]: minor: "theconstructor" -> "the constructor"

File: linalg/amgxsolver.hpp [hunk 20]
```diff
@@ -71,85 +71,122 @@ class AmgXSolver : public Solver
...
-      Sets Operator for AmgX library, either MFEM SparseMatrix or HypreParMatrix
+   /** @brief  Sets the Operator that is going to be solved via AmgX.
+      Supports operators based on either an MFEM SparseMatrix or
+      HypreParMatrix.
    */
    virtual void SetOperator(const Operator &op);
 
-   /**
-      Replaces the matrix coefficients in the AmgX solver.
+   /** @brief Change the input operator that is being solved via AmgX.
```
Comment (paul-hilscher) [inline linalg/amgxsolver.hpp:162]: minor: Changes ?

File: linalg/amgxsolver.hpp [hunk 21]
```diff
@@ -71,85 +71,122 @@ class AmgXSolver : public Solver
...
+       @a x vector will be set to zero, otherwise the value of @a x passed
+       in will be used.
+   */
    virtual void Mult(const Vector& b, Vector& x) const;
 
+   /** @brief Return the number of iterations that were executed during the
+       last solve phase. */
    int GetNumIterations();
 
+   /** @brief Read in the Amgx parameters either through a file or directly
```
Comment (paul-hilscher) [inline linalg/amgxsolver.hpp:181]: minor: "Amgx" -> "AmgX"

File: linalg/amgxsolver.hpp (hunk 18 above)
Comment (paul-hilscher) [inline linalg/amgxsolver.hpp:147]: minor: "numberof" -> "number of"

File: linalg/amgxsolver.hpp (hunk 19 above)
Comment (paul-hilscher) [inline linalg/amgxsolver.hpp:148]: minor: "theconstructor" -> "the constructor"

File: linalg/amgxsolver.hpp [hunk 22]
```diff
@@ -71,85 +71,122 @@ class AmgXSolver : public Solver
...
 
-   /**
-      Once the solver configuration has been established, either through the
-      ReadParameters method, InitMPITeams will initialize the library and create
-      MPI teams based on the number of devices on each node (nDevs). If
-      configuring with constructor, the constructor will make this call.
+   /** @brief Initialize the AmgX library and create MPI teams based on the
+      numberof devices on each node @a nDevs.  If configuring with a
+      constructor, theconstructor will make this call, otherwise this will need
+      to be calledafter the solver configuration has been established through
```
Comment (paul-hilscher) [inline linalg/amgxsolver.hpp:149]: minor: "calledafter" -> "called after"

File: linalg/amgxsolver.hpp (hunk 21 above)
Comment (paul-hilscher) [inline linalg/amgxsolver.hpp:181]: minor: "Amgx" -> "AmgX"

File: linalg/amgxsolver.hpp [hunk 23]
```diff
@@ -160,16 +197,19 @@ class AmgXSolver : public Solver
       two iterations of an AMG V cycle with AmgX's default smoother (block
       Jacobi).
 
-      As a solver the preconditioned conjugate gradient method is used. The AMG
-      V-cycle with a block Jacobi smoother is used as a preconditioner.
+      When configured as a solver the preconditioned conjugate gradient method
+      is used with the AMG V-cycle with a block Jacobi smoother is used as a
```
Comment (paul-hilscher) [inline linalg/amgxsolver.hpp:201]: minor: I find this a bit hard to read. Maybe "with the AMG V-cycle with a block Jacobi smoother is used as a preconditioner" ->  "with the AMG V-cycle **and** a block Jacobi smoother as preconditioner." ?

File: linalg/amgxsolver.hpp [hunk 24]
```diff
@@ -181,36 +221,43 @@ class AmgXSolver : public Solver
...
-                                 const Array<int> &loc_I, const Array<int64_t> &loc_J,
+                                 const Array<int> &loc_I,
+                                 const Array<int64_t> &loc_J,
                                  const bool update_mat = false);
 
-   // Consolidates matrix diagonal and off diagonal data for all ranks in an MPI
-   // team. Root rank of each MPI team holds the the consolidated data and sets
-   // matrix.
+   /** @brief Consolidates matrix diagonal and off diagonal data for all ranks
+       in an MPIteam. Root rank of each MPI team holds the the consolidated
```
Comment (paul-hilscher) [inline linalg/amgxsolver.hpp:234]: minor: "MPIteam" -> "MPI team" .

File: linalg/slepc.hpp [hunk 25]
```diff
@@ -69,39 +77,105 @@ class SlepcEigenSolver
...
+   */
    enum SpectralTransformation
    {
+      /// Utilize the shift of origin strategy
       SHIFT,
+      /// Utilize the shift and invert strategy
       SHIFT_INVERT
    };
 
+   /** @brief Set the which eigenvalues the solver will target and the order
```
Comment (paul-hilscher) [inline linalg/slepc.hpp:160]: minor: "Set the which" -> "Set which"

File: linalg/solvers.hpp [hunk 26]
```diff
@@ -338,7 +344,11 @@ class OperatorJacobiSmoother : public Solver
    /// Replace diagonal entries with their absolute values.
    void SetPositiveDiagonal(bool pos_diag = true) { use_abs_diag = pos_diag; }
 
+   /// Approach the solution of the linear system by applying jacobi smoothing.
    void Mult(const Vector &x, Vector &y) const;
+
+   /** @brief Approach the solution of the transposed linear system by applying
+       jacobi smoothing. */
```
Comment (paul-hilscher) [inline linalg/solvers.hpp:351]: minor/pendantic: "jacobi" -> "Jacobi"

File: linalg/strumpack.hpp [hunk 27]
```diff
@@ -31,30 +31,37 @@ namespace mfem
...
-       format. All data is copied so the original matrix may be deleted. */
+   /** @brief Creates a copy of the parallel matrix hypParMat in STRUMPACK's
+       RowLoc format.
+
+       All data is copied so the original matrix may be deleted. */
    STRUMPACKRowLocMatrix(const Operator &op, bool sym_sparse = false);
 
    ~STRUMPACKRowLocMatrix();
 
+   /// Matrix vector products are not supported on for this type of matrix.
```
Comment (paul-hilscher) [inline linalg/strumpack.hpp:54]: minor:  "not supported on for this" -> "not supported for this" ?

**Comment by paul-hilscher** [comment]
> @paul-hilscher Alright I addressed all of the things you found and did a little cleanup as well. @mlstowell, can you have a look at this sometime this week?

Thanks @acfisher for the improvements. I have added a few additional minor comments. Could you please have a look.

**Review by paul-hilscher — APPROVED** [review]
Left few minor comments. Once addressed, all good from my side.

**Comment by acfisher** [comment]
@paul-hilscher Thanks for the additional series of copy edits here.  It's really pushing the quality of these comments up.  It looks like my spacebar wasn't operating for some of this work ;).

**Comment by acfisher** [comment]
@chakshinglee do you think you might have some time to take a look at this PR in the next week?

File: linalg/solvers.hpp [hunk 28]
```diff
@@ -338,7 +344,11 @@ class OperatorJacobiSmoother : public Solver
    /// Replace diagonal entries with their absolute values.
    void SetPositiveDiagonal(bool pos_diag = true) { use_abs_diag = pos_diag; }
 
+   /// Approach the solution of the linear system by applying jacobi smoothing.
```
Comment (vladotomov) [inline linalg/solvers.hpp:347]: `Jacobi`

**Review by vladotomov — APPROVED** [review]
Thanks for taking the time for this Aaron.

File: linalg/amgxsolver.hpp (hunk 3 above)
Comment (vladotomov) [reply linalg/amgxsolver.hpp:138]: :exclamation:

File: linalg/solvers.hpp (hunk 28 above)
Comment (acfisher) [reply linalg/solvers.hpp]: Good catch.

**Comment by acfisher** [comment]
@vladotomov Thanks for taking the time to review this.

**Review by tzanio — APPROVED** [review]

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Pull request merged**